admin-api-server/test/app.e2e.spec.ts

```diff
@@ -2035,11 +2038,12 @@ describe('AppController (e2e)', () => {
         .set('authorization', bearer)
         .query({ resolution: 'hour' });
       expect(res.statusCode).toEqual(200);
+      timestamps = AnalyticsService.prototype.getTimestampData(
```

Comment:
It's not good to define expected values in test cases based on execution of the code that is under construction.

Expected data should be hardcoded here instead.

(otherwise, if a bug is introduced in this shared code that the test relies on to figure out what it expects, tests might succeed but only because its expectations are now flawed too.

Reply:
Clever though. Thanks

Reply:
All new test cases use static test data now.